From 7eb428f1dfa5bb317981c60be449b6baeba69a07 Mon Sep 17 00:00:00 2001
From: Dongjin Kim <tobetter@gmail.com>
Date: Thu, 27 Feb 2020 11:50:05 +0900
Subject: [PATCH 052/109] ODROID-XU4: FIXME: Revert "regulator: core: Let
 boot-on regulators be powered off"

FIXME: this patch crashes when Mali is activated

This reverts commit f44b07472f29ae313ce875dc7b9c75b100c608b8.

Change-Id: I2f3e8e68b3172c2c1d035032b47fb256c5757be8
---
 drivers/regulator/core.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/regulator/core.c b/drivers/regulator/core.c
index 16114aea099a..41d0a739ceaf 100644
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -1436,9 +1436,7 @@ static int set_machine_constraints(struct regulator_dev *rdev)
 			rdev_err(rdev, "failed to enable: %pe\n", ERR_PTR(ret));
 			return ret;
 		}
-
-		if (rdev->constraints->always_on)
-			rdev->use_count++;
+		rdev->use_count++;
 	}
 
 	print_constraints(rdev);
-- 
2.25.1

